### PR TITLE
add declaration file (index.d.ts)

### PR DESCRIPTION
**Pull request by wiloke1**
#### Adding declaration file helps editor suggest props for you and report errors typescript

Screenshot 2019-12-19 00 21 23
Screenshot 2019-12-19 00 49 38
Screenshot 2019-12-19 10 34 11
Screenshot 2019-12-19 12 15 06
Screenshot 2019-12-19 12 15 28

### CODE REVIEW

**Comment by palingheorghe** [comment]
Thank you for this PR! This will sure come in handy for our users! I'll have to review it and then add it in Galio's next version. This will probably happen after New Year Eve. 

You've done a great help to our community ❤️!

**Comment by wiloke1** [comment]
@palingheorghe I don't know icon font type is correct or not? But some fonts are not working
I need exactly the type of icon font family. Can you give it to me?
https://github.com/wiloke1/galio/blob/ba180cc1aa5e35dc62c7d6ad2210f1a0544d0198/src/index.d.ts#L12-L27

**Comment by palingheorghe** [comment]
Hi @wiloke1 ! You can find more about the font family type names here: https://github.com/galio-org/galio/blob/master/src/helpers/getIconType.js . These must be: `zocial, octicon, material, material-community, ionicon, foundation, evilicons, entypo, font-awesome, font-awesome-5, simple-line-icon, feather, antdesign`. On top of that we have `Galio` as one of the font family names.

**Comment by superandrew** [comment]
👍 adding myself to that, is this going to be released anytime soon?

**Comment by AndyOsei** [comment]
We are working on releasing hotfixes for immediate pending issues, will discuss on considering adding this pull request to that list with the others after review...

Am confident it will be added pretty soon...

**Comment by kevbradwick** [comment]
just wanted to add my support for this PR - I could really do with it right now 👍

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    uppercase?: boolean;
+  }
+  export class Button extends React.Component<ButtonProps> {}
+
+  export interface CardProps extends BaseProps {
+    card?: boolean;
+    shadow?: boolean;
+    borderless?: boolean;
+    image?: string;
+    imageBlockStyle?: strubg;
```
Comment (yasintz) [inline src/index.d.ts:83]: what is `strubg`?

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+      >,
+      BaseProps {
+    type?: TextInputProps['keyboardType'];
+    password?: boolean;
+    label?: string;
+    bgColor?: string;
+    rounded?: boolean;
+    borderless?: boolean;
+    viewPass?: boolean;
+    icon?: string;
```
Comment (yasintz) [inline src/index.d.ts:147]: Can you add the `iconContent` prop?
https://github.com/galio-org/galio/blob/master/src/Input.js#L152

**Review by byeze — APPROVED** [review]

**Comment by SapioBeasley** [comment]
merge please .. please? lol

**Pull request merged**